### ISSUE TITLE
SVG that is aria-hidden, cannot query by role using title for accessible name

### ISSUE THREAD

**Issue by Josh68** [testing-library/dom-testing-library#1132]
- `@testing-library/dom` version: 7.31.2
- Testing Framework and version: Jest, 27.4.6
  
- DOM Environment: jsdom, 27.4.6
  

### Relevant code or config:

In React, but using `dom-testing-library` under the hood:

```js
// code
const MySvg = () => <svg role="img" aria-hidden="true"...><title>svg_name</title><path>...</path></svg>;

// test, fails
render(<MySvg />);
screen.getByRole("img", { hidden: true, name: "svg_name" }));
```

Note that this will still get the SVG, if it's the only element with `role="img"`

```js
screen.getByRole("img", { hidden: true }));
```

SVG is not `aria-hidden`:

```js
// code
const MySvg = () => <svg role="img" ...><title>svg_name</title><path>...</path></svg>;

// test, passes
render(<MySvg />);
screen.getByRole("img", { name: "svg_name" }));
```

### What you did:

Tried to test rendering of an `aria-hidden` `svg` by using `getByRole` with the options `{ hidden: true, name: accessible_name }` where `accessible_name` is the text node inside the `svg`'s `title` element.

### What happened:

Found I could only make this work when the `svg` is not `aria-hidden` (and I don't need to add `{ hidden: true }` to the `getByRole` options, although the test passes whether or not I remove that option).

### Reproduction:

https://codesandbox.io/s/happy-darkness-zy83ig (run tests)

### Problem description:

[This PR](https://github.com/testing-library/dom-testing-library/issues/685) evidently fixed using `title` as an accessible name for `svg` elements. But evidently this still doesn't work if the `svg` is `aria-hidden`.
